control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -1534,6 +1673,140 @@ func TestReconcileDeleteEndpointWithNamespaces(t *testing.T) {
+						token, _, err = consulClient.ACL().Login(&api.ACLLoginParams{
+							AuthMethod:  test.AuthMethod,
+							BearerToken: test.ServiceAccountJWTToken,
+							Meta: map[string]string{
+								TokenMetaPodNameKey: fmt.Sprintf("%s/%s", svc.Meta[MetaKeyKubeNS], svc.Meta[MetaKeyPodName]),
+							},
+						}, &writeOpts)
+
+						require.NoError(t, err)
+					}
+				}
+
+				// Create the endpoints controller.
+				ep := &EndpointsController{
+					Client:                     fakeClient,
+					Log:                        logrtest.TestLogger{T: t},
+					ConsulClient:               consulClient,
+					AllowK8sNamespacesSet:      mapset.NewSetWith("*"),
+					DenyK8sNamespacesSet:       mapset.NewSetWith(),
+					ReleaseName:                "consul",
+					ReleaseNamespace:           "default",
+					EnableConsulNamespaces:     true,
+					EnableNSMirroring:          ts.Mirror,
+					NSMirroringPrefix:          ts.MirrorPrefix,
+					ConsulDestinationNamespace: ts.DestConsulNS,
+				}
+				if tt.enableACLs {
+					ep.AuthMethod = test.AuthMethod
+				}
+
+				// Set up the Endpoint that will be reconciled, and reconcile.
+				namespacedName := types.NamespacedName{
+					Namespace: ts.SourceKubeNS,
+					Name:      "service-deleted",
+				}
+				resp, err := ep.Reconcile(context.Background(), ctrl.Request{
+					NamespacedName: namespacedName,
+				})
+				require.NoError(t, err)
+				require.False(t, resp.Requeue)
+
+				cfg.Namespace = ts.ExpConsulNS
+				consulClient, err = api.NewClient(cfg)
+				require.NoError(t, err)
+
+				// After reconciliation, Consul should not have any instances of service-deleted.
+				serviceInstances, _, err := consulClient.Catalog().Service(tt.consulSvcName, "", &api.QueryOptions{Namespace: ts.ExpConsulNS})
+				require.NoError(t, err)
+				require.Empty(t, serviceInstances)
+				proxyServiceInstances, _, err := consulClient.Catalog().Service(fmt.Sprintf("%s-sidecar-proxy", tt.consulSvcName), "", &api.QueryOptions{Namespace: ts.ExpConsulNS})
+				require.NoError(t, err)
+				require.Empty(t, proxyServiceInstances)
+
+				if tt.enableACLs {
+					_, _, err = consulClient.ACL().TokenRead(token.AccessorID, nil)
+					require.EqualError(t, err, "Unexpected response code: 403 (ACL not found)")
+				}
+			})
+		}
+	}
+}
+
+// Tests deleting an Endpoints object, with and without matching Consul and K8s service names when Consul namespaces are enabled.
+// This test covers EndpointsController.deregisterService when the map is nil (not selectively deregistered).
+func TestReconcileDeleteGatewayWithNamespaces(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		ConsulNS string
+	}{
+		"default Consul namespace": {
+			ConsulNS: "default",
+		},
+		"other Consul namespace": {
+			ConsulNS: "other",
+		},
+	}
+	for name, ts := range cases {
+		cases := []struct {
+			name              string
+			consulSvcName     string
```

Comment:
It looks like `consulSvcName` is always the same between test cases so we can remove it.